### ISSUE TITLE
IPv6 support

### ISSUE THREAD

**Issue by JanC** [GlennChiu/GCNetworkReachability#9]
Hi,
does this actually work on IPv6? I see that you use a `sockaddr_in` instead of `sockaddr_in6` in [GCNetworkReachabilitySetSocketAddress](https://github.com/GlennChiu/GCNetworkReachability/blob/master/GCNetworkReachability.m#L118)

The IPv4 structs should not be used based on what Apple said in WWDC (session "Your App and Next Generation Networks")

![ipv6](https://cloud.githubusercontent.com/assets/707703/9220500/509b3268-40e3-11e5-97c7-5abd49ed166f.png)

**Comment by Coeur** [comment]
You are right, there is a bug here if trying to monitor an ipv6 address, as `inet_pton()` requires allocating more space in that case.

But as GCNetworkReachability is a dead project...
